src/main/java/frc/robot/GenericPID/PolynomialSegment.java

```diff
@@ -0,0 +1,60 @@
+package frc.robot.GenericPID;
+
+import java.util.ArrayList;
+
+public class PolynomialSegment extends PathSegmentBase {
+    ArrayList<Double> coeffs; //the index represents the x degree, so [1,2,3] = 3x^2 + 2x + 1
+    public double y(double x) {
+        double ret = 0;
+        for (int i = 0; i < coeffs.size(); i++) {
+            ret += coeffs.get(i) * Math.pow(x, i);
+        }
+        return ret;
+    }
+    public double derivative(double x) {
```

Comment:
override it! even if it doesn't have an implementation in the parent class

Reply:
it's an abstract method, that's overriding (abstract classes are like a mix of an interface and a parent class)